### PR TITLE
igmvbuilder: support VSM isolation platform type

### PR DESCRIPTION
**Pull request by msft-jlange**
This change enables building an IGVM file that support the `VSM_ISOLATION` platform type.

### CODE REVIEW

File: igvmbuilder/src/cmd_options.rs [hunk 1]
```diff
@@ -64,6 +64,10 @@ pub struct CmdOptions {
     #[arg(long, default_value_t = false)]
     pub native: bool,
 
+    /// Includ VSM isolation platform target.
```
Comment (roy-hopkins) [inline igvmbuilder/src/cmd_options.rs:67]: Typo: "Includ"

File: igvmbuilder/src/vmsa.rs [hunk 2]
```diff
@@ -52,56 +176,105 @@ pub fn construct_vmsa(
...
+                vmsa.rflags = *r;
+            }
+            X86Register::Idtr(table) => {
+                vmsa.gdtr.base = table.base;
+                vmsa.gdtr.limit = table.limit as u32;
+            }
+            X86Register::Gdtr(table) => {
+                vmsa.idtr.base = table.base;
+                vmsa.idtr.limit = table.limit as u32;
+            }
```
Comment (roy-hopkins) [inline igvmbuilder/src/vmsa.rs:219]: These two have been mixed up - Idtr sets gdtr and vise versa.

File: igvmbuilder/src/vmsa.rs [hunk 3]
```diff
@@ -52,56 +176,105 @@ pub fn construct_vmsa(
...
+            X86Register::Rsp(r) => {
+                vmsa.rsp = *r;
+            }
+            X86Register::Rbp(r) => {
+                vmsa.rbp = *r;
+            }
+            X86Register::Rsi(r) => {
+                vmsa.rsi = *r;
+            }
+            X86Register::R8(r) => {
```
Comment (roy-hopkins) [inline igvmbuilder/src/vmsa.rs:191]: No rdi?

File: igvmbuilder/src/vmsa.rs [hunk 4]
```diff
@@ -4,46 +4,170 @@
...
+            X86Register::Rsp(r) => {
+                context.rsp = *r;
+            }
+            X86Register::Rbp(r) => {
+                context.rbp = *r;
+            }
+            X86Register::Rsi(r) => {
+                context.rsi = *r;
+            }
+            X86Register::R8(r) => {
```
Comment (roy-hopkins) [inline igvmbuilder/src/vmsa.rs:72]: No rdi?

**Review by roy-hopkins — CHANGES_REQUESTED** [review]

File: igvmbuilder/src/vmsa.rs (hunk 2 above)
Comment (msft-jlange) [reply igvmbuilder/src/vmsa.rs:219]: Oops!

File: igvmbuilder/src/vmsa.rs (hunk 3 above)
Comment (msft-jlange) [reply igvmbuilder/src/vmsa.rs:191]: No, `Rdi` is not a member of `Register::X86Register` in the `igvm` crate.

**Review by roy-hopkins — APPROVED** [review]
Looks good, and tested the QEMU igvm output on SEV-SNP - all works.

**Review by stefano-garzarella — APPROVED** [review]
Tested QEMU/KVM IGVM and launch measurement.

**Pull request merged**